lock.go

```diff
@@ -13,35 +14,52 @@ import (
 const luaRefresh = `if redis.call("get", KEYS[1]) == ARGV[1] then return redis.call("pexpire", KEYS[1], ARGV[2]) else return 0 end`
 const luaRelease = `if redis.call("get", KEYS[1]) == ARGV[1] then return redis.call("del", KEYS[1]) else return 0 end`
 
+var ErrCanntGetLock = errors.New("can't get lock")
+
 // RedisClient is a minimal client interface
 type RedisClient interface {
 	SetNX(key string, value interface{}, expiration time.Duration) *redis.BoolCmd
 	Eval(script string, keys []string, args ...interface{}) *redis.Cmd
 }
 
+// Handler is method wrapper
+type Handler func() error
+
 // Lock allows distributed locking
 type Lock struct {
 	client RedisClient
 	key    string
-	opts   LockOptions
+	opts   Options
 
 	token string
 	mutex sync.Mutex
 }
 
+// RunWithLock run some code with Redis Lock
+func RunWithLock(client RedisClient, key string, handler Handler, opts *Options) error {
+	locker, err := ObtainLock(client, key, opts)
+	if err != nil {
+		return err
+	}
+	defer locker.Unlock()
+	return handler()
+}
+
 // ObtainLock is a shortcut for NewLock().Lock()
-func ObtainLock(client RedisClient, key string, opts *LockOptions) (*Lock, error) {
-	lock := NewLock(client, key, opts)
-	if ok, err := lock.Lock(); err != nil || !ok {
+func ObtainLock(client RedisClient, key string, opts *Options) (*Lock, error) {
+	locker := NewLock(client, key, opts)
+	if ok, err := locker.Lock(); err != nil {
 		return nil, err
+	} else if !ok {
+		return nil, ErrCanntGetLock
 	}
-	return lock, nil
+	return locker, nil
 }
 
 // NewLock creates a new distributed lock on key
-func NewLock(client RedisClient, key string, opts *LockOptions) *Lock {
+func NewLock(client RedisClient, key string, opts *Options) *Lock {
```

Comment:
for the new major version, I would like to rename this to just `New`

Reply:
ok, I can do this